README.md

```diff
@@ -73,14 +73,25 @@ This diagram covers the major classes of transformations. The most basic ones ar
 Here is a very simple example of an algebra (`eval`) and how to apply it to a recursive structure.
 
 ```scala
+
+// we will need a Functor[Expr] in order to call embedT bellow
+implicit val exprFunctor = new scalaz.Functor[Expr] {
+  override def map[A, B](fa: Expr[A])(f: (A) => B) = fa match{
+    case Num(value) => Num[B](value)
+    case Mul(l, r) => Mul(f(l), f(r))
+  }
+}
+
 val eval: Algebra[Expr, Long] = { // i.e. Expr[Long] => Long
   case Num(x)    => x
   case Mul(x, y) => x * y
 }
+ 
+def someExpr[T[_[_]]: CorecursiveT]: T[Expr] =
+  Mul(Num[T[Expr]](2).embedT, Mul(Num[T[Expr]](3).embedT,
+      Num[T[Expr]](4).embedT).embedT).embedT
```

Comment:
Glad you were able to figure out how to fix it. This definitely works, although a more general version is
```scala
def someExpr[T](implicit T: Corecursive.Aux[T, Expr]): T =
  Mul(Num[T](2).embed, Mul(Num[T](3).embed,
      Num[T](4).embed).embed).embed

someExpr[Mu[Expr]].cata(eval)
```
The `T[_[_]]` type is more restrictive – basically only allowing Fix/Mu/Nu, because it needs a type that can be written with a functor-shaped hole. Whereas `T` (with the `Corecursive.Aux[T, Expr]` implicit) is any proper type from which you can _extract_ an Expr functor – that clearly includes `Fix[Expr]`, but could also include, say, a directly-recursive `ExprR` that had a mapping to the `Expr` functor (note that `ExprR` is a proper type that doesn’t have a type parameter for `Expr` anywhere, so the `T[_[_]]` couldn’t match it).

Also, with an upcoming PR, even the `T[_[_]]` version would allow you to just write `.embed` rather than `.embedT`.

I eventually want to get rid of the `CorecursiveT` and other `*T` type classes, but at least until we have mutual recursion they are still useful.